mitodl/bootcamp-ecommerce
# feat: use poetry dependency management

#### What are the relevant tickets?
https://github.com/mitodl/bootcamp-ecommerce/issues/1440

#### What's this PR do?
- Adds support for poetry dependency management
- updates GitHub actions to use poetry
- updates docker to use poetry

#### How should this be manually tested?
- The build should succeed
- The tests should pass
- The `pyproject.toml` dependencies should be accurate as per `^` (For main dependencies only minor and patch versions should be upgraded in this PR)

**Try two build scenarios:**
- Run this PR on an existing build from master, ideally `docker-compose build --no-cache` should work
- Run a fresh instance `docker-compose down`, `docker-compose build --no-cache`, `docker-compose up`

**Try the Functionality test scenarios**
- Try creating new account/Signin
- Try setting up bootcamp in CMS and make sure it displays and works the same as before
- Try running commands like `pytest, pylint` and they should work
- Try running one or more management commands
- It would be a good idea to test If you have Hubspot setup locally
